### ISSUE TITLE
model_diff for Stan models

### ISSUE THREAD

**Issue by seth127** [metrumresearchgroup/bbr#341]
# Summary
As a user, I would like to be able to use the `model_diff()` function to compare the following files between Stan models: `<run>.stan`, `<run>-standata.R`, `<run>-init.R`, `<run>-stanargs.R`.

# Tests
- tests/testthat/test-model-diff.R
  - model_diff.bbi_stan_model happy path based_on
  - model_diff.bbi_stan_model happy path .mod2 arg
  - model_diff.bbi_stan_model works with other files

**Comment by seth127** [comment]
Something like this (haven't tested, but this is the design idea)
```r
# add this to the top
#' @param .file ... For `bbi_stan_model` objects, `"model"` (the default) will compare the 
#' `.stan` files. 

#' @export
model_diff.bbi_stan_model <- function(
  .mod,
  .mod2 = NULL,
  .file = c("model", "standata", "init", "stanargs"),
  ...
) {
  .file <- match.arg(.file)

  .mod2 <- model_diff_get_comp(.mod, .mod2)

  comp_files <- map(list(.mod, .mod2), function(.m) {
    if (.file == "model") {
      return(build_path_from_model(.m, ".stan"))
    } else {
      return(build_path_from_model(.m, glue("-{.file}.R")))
    }
  })

  do.call(model_diff_impl, comp_files)
}
```

**Comment by seth127** [comment]
Closed by #371
